packages/slice-machine/lib/mock/Slice.ts

```diff
@@ -1,53 +1,80 @@
-import { Models } from "@slicemachine/core";
-import * as Widgets from "./misc/widgets";
-import { snakelize } from "../utils/str";
-import { handleFields } from "./misc/handlers";
-import { FieldsSM } from "@slicemachine/core/build/models/Fields";
-import { SliceSM, VariationSM } from "@slicemachine/core/build/models/Slice";
+import {
+  SharedSlice,
+  SlicesTypes,
+  Variation,
+} from "@prismicio/types-internal/lib/customtypes/widgets/slices";
+import {
+  SharedSliceMockConfig,
+  VariationMockConfig,
+  generateSliceMock,
+  renderSliceMock,
+} from "@prismicio/mocks";
+import { buildFieldsMockConfig } from "./LegacyMockConfig";
+import { Slices, SliceSM } from "@slicemachine/core/build/models/Slice";
 
-const createEmptyMock = (
-  sliceName: string,
-  variation: VariationSM
-): Models.VariationMock => ({
-  variation: variation.id,
-  name: variation.name,
-  slice_type: snakelize(sliceName),
-  items: [],
-  primary: {},
-});
-
-export default function MockSlice(
-  sliceName: string,
-  model: SliceSM,
-  mockConfig: Record<string, Record<string, Record<string, unknown>>> // not sure about this one.
-): Models.SliceMock {
-  const variations = model.variations.map((variation) => {
-    const mock = createEmptyMock(sliceName, variation);
-    const handler: (
-      fields?: FieldsSM,
-      mocks?: Record<string, unknown>
-    ) => Models.VariationMock["primary"] = handleFields(Widgets);
-
-    const maybeMockConfig = mockConfig?.[variation.id]?.primary || {};
-    mock.primary = handler(variation.primary, maybeMockConfig);
-
-    const items: Models.VariationMock["items"] = [];
+function buildVariationMockConfig(
+  model: Variation,
+  legacyVariationMockConfig?: Partial<Record<string, Record<string, unknown>>>
+): VariationMockConfig {
+  const primaryFields =
+    model.primary &&
+    buildFieldsMockConfig(
+      model.primary,
+      legacyVariationMockConfig?.["primary"]
+    );
+  const itemFields =
+    model.items &&
+    buildFieldsMockConfig(model.items, legacyVariationMockConfig?.["items"]);
+  return {
+    nbItemsBlocks: 1,
+    primaryFields,
+    itemFields,
+  };
+}
 
-    const repeat = variation.items;
-    if (!repeat || (repeat && repeat.length === 0)) {
+export function buildSliceMockConfig(
+  model: SharedSlice,
+  legacyMockConfig: Partial<
```

Comment:
what a weird type ^^
Can they be extracted all the legacyMockConfig type for more clarity ?

Reply:
what do you mean? create a type alias for just this line?

Reply:
I mean typing all the different LegacyMocks so that they can be re-used (either in the UI or anywhere else).

Reply:
They are typed already partly done thanks to the new validators but I don't see the benefit of doing more here.
It wasn't typed before and now we have a contract for each field to support the library.
I don't get why we should create types for some code that is meant to disappear?
Especially in the context of this PR that has a goal of getting rid of this old thing.
If you end up deciding that we'll keep the current mock config, then definitely you'll need to improve the typing but that's not something we should focus right now again.